### PR TITLE
Add scripts table

### PR DESCRIPTION
**Pull request by squidsoup**
## Done
* Display commissioning and testing scripts in tables.
* Display script source on expand.

## QA
* Ensure you have some testing scripts (you may need to upload some junk via old maas settings).
* Visit /scripts/testing and /scripts/commissioning
* Ensure sorting works, ensure searching works
* Click script name, you should see the script source in a code block.

### CODE REVIEW

**Comment by squidsoup** [comment]
@lilyvidenova unfortunately there is no "author" in the script object, so I've dropped that column for now. Shall I file a MAAS bug requesting that the API is updated?

File: ui/src/app/settings/views/Scripts/ScriptsList.js [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+          role: "rowheader"
+        },
+        {
+          content: script.description
+        },
+        { content: uploadedOn },
+        {
+          content: (
+            <>
+              <span className="p-tooltip p-tooltip--left">
```
Comment (huwshimi) [inline ui/src/app/settings/views/Scripts/ScriptsList.js]: This tooltip span doesn't actually need to be there, looks like we've accidentally copied that in a few places.

File: ui/src/app/settings/views/Scripts/ScriptsList.js [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+              content: "Script name",
+              sortKey: "name"
+            },
+            {
+              content: "Description",
+              sortKey: "description"
+            },
+            {
+              content: "Uploaded on",
+              sortKey: "uploaded_on"
```
Comment (huwshimi) [inline ui/src/app/settings/views/Scripts/ScriptsList.js]: The sort data has a key of `uploadedOn`.

File: ui/src/app/settings/views/Scripts/ScriptsList.scss [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    flex: 0 0 45%;
+  }
+
+  // Uploaded on
+  &:nth-child(3) {
+    flex: 0 0 15%;
+  }
+
+  // Last edited
+  &:nth-child(4) {
```
Comment (huwshimi) [inline ui/src/app/settings/views/Scripts/ScriptsList.scss]: This looks like it's actually the actions column, setting the width for this one might not be necessary.

File: ui/src/app/settings/views/Scripts/ScriptsList.test.js [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+              }
+            ]
+          }
+        ]
+      }
+    };
+  });
+
+  it("displays a loading component if loading", () => {
+    const state = { ...initialState };
```
Comment (huwshimi) [inline ui/src/app/settings/views/Scripts/ScriptsList.test.js]: I'm not sure it's necessary to shallow copy in these tests as `initialState` is reset between tests anyway.

**Comment by lyubomir-popov** [comment]
Hi could you please point me to some scripts for both categories, I tried creating an empty file and uploading it via the old maas but it didn't appear in the new one.

**Comment by squidsoup** [comment]
@lilyvidenova yes, you need to refresh the page, as this uses the http API not websockets (there are no realtime updates like other parts of the app).

**Comment by squidsoup** [comment]
@lyubomir-popov @lilyvidenova can you check that you have:

```
REACT_APP_BASE_URL="http://<your-maas-ip>:5240/
```
in your `.env.development.local` please

**Comment by lyubomir-popov** [comment]
@squidsoup that fixed it, thanks!

**Comment by lyubomir-popov** [comment]
Design looks good, just one comment if possible - 
Ideally this focus state should be on the chevron button even if the whole area acts as a click target.
image

**Comment by lyubomir-popov** [comment]
One more, for a future enhancement: Once you have opened a script, it's a long scroll buck to the top of the table / row you expanded. Would be nice if we had a close mechanism at the bottom of an expanded row:

image

**Comment by squidsoup** [comment]
> One more, for a future enhancement: Once you have opened a script, it's a long scroll back to the top of the table / row you expanded. Would be nice if we had a close mechanism at the bottom of an expanded row:

Good suggestion, thanks @lyubomir-popov - have added a card https://github.com/canonical-web-and-design/maas-ui/issues/189

**Comment by squidsoup** [comment]
> Design looks good, just one comment if possible -
> Ideally this focus state should be on the chevron button even if the whole area acts as a click target.
> image

@lyubomir-popov I've created an issue for consolidating the button toggles in both DhcpList and ScriptsList here into a component here https://github.com/canonical-web-and-design/maas-ui/issues/190

Perhaps we could address the focus state at the same time there. I've added a question for you on the issue.

**Pull request merged**